pkg/cache/testsuite/strings.go

```diff
@@ -0,0 +1,179 @@
+// Copyright © 2020 by PACE Telematics GmbH. All rights reserved.
+// Created at 2020/08/12 by Marius Neugebauer
+
+package testsuite
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pace/bricks/maintenance/log"
+	"github.com/pace/bricks/pkg/cache"
+	"github.com/stretchr/testify/suite"
+)
+
+type StringsTestSuite struct {
+	suite.Suite
+	Cache cache.Strings
+}
+
+func (suite *StringsTestSuite) TestSetString() {
+	c := suite.Cache
+	ctx := log.WithContext(context.Background())
+
+	_ = c.Forget(ctx, "foo") // make sure it doesn't exist
+	suite.Run("does not error", func() {
+		err := c.SetString(ctx, "foo", "bar", time.Second)
+		suite.NoError(err)
+	})
+	_ = c.Forget(ctx, "foo") // clean up
+
+	suite.Run("does not error when repeated", func() {
+		_ = c.SetString(ctx, "foo", "bar", time.Second)
+		err := c.SetString(ctx, "foo", "bar", time.Second)
+		suite.NoError(err)
+	})
+	_ = c.Forget(ctx, "foo") // clean up
+
+	suite.Run("stores a value", func() {
+		_ = c.SetString(ctx, "foo", "bar", 0)
+		value, _, _ := c.GetString(ctx, "foo")
+		suite.Equal("bar", value)
+	})
+
+	for i := 0; i <= 5; i++ { // make sure it doesn't exist
+		_ = c.Forget(ctx, fmt.Sprintf("foo%d", i))
+	}
+	suite.Run("does not error on simultaneous use", func() {
+		var wg sync.WaitGroup
+		for i := 0; i <= 5; i++ {
+			wg.Add(1)
+			go func() {
+				err := c.SetString(ctx, fmt.Sprintf("foo%d", i), "bar", 0)
+				suite.NoError(err)
+				wg.Done()
+			}()
+			wg.Wait()
+		}
+	})
+	for i := 0; i <= 5; i++ { // clean up
+		_ = c.Forget(ctx, fmt.Sprintf("foo%d", i))
+	}
+}
+
+func (suite *StringsTestSuite) TestGetString() {
+	c := suite.Cache
+	ctx := log.WithContext(context.Background())
+
+	_ = c.Forget(ctx, "foo") // make sure it doesn't exist
+	suite.Run("returns the ttl if set", func() {
+		_ = c.SetString(ctx, "foo", "bar", time.Minute)
```

Comment:
I should use different keys in every test that can run synchronously.

Reply:
No need, testify runs everything in sequence unless requested otherwise. See https://github.com/stretchr/testify/issues/187.